### PR TITLE
python3Packages.meraki: init at 1.30.0

### LINKED ISSUE #222267: Package request: meraki
**Project description**

Cisco Meraki Python module

**Metadata**

* homepage URL: https://pypi.org/project/meraki/
* source URL: https://github.com/meraki/dashboard-api-python/
* license: mit
* platforms: unix

### PR DESCRIPTION
**Pull request by dylanmtaylor**
###### Description of changes

Add Meraki Python development API library. This closes #222267.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by dylanmtaylor** [comment]
Got it building now :) 
image

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -9295,6 +9295,8 @@ with pkgs;
 
   memtester = callPackage ../tools/system/memtester { };
 
+  meraki = python3Packages.callPackage ../development/python-modules/meraki { };
```
Comment (lilyinstarlight) [inline pkgs/top-level/all-packages.nix:9298]: Put this in `pkgs/top-level/python-modules.nix` instead. If you really want it at the top-level pkgs (e.g. if it provides a CLI), than you can `inherit (python3Packages) meraki;` here after adding it to the other file

File: pkgs/development/python-modules/meraki/default.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+, buildPythonPackage
+, fetchPypi
+, python3
+}:
+
+with python3.pkgs;
+
+buildPythonPackage rec {
+  pname = "meraki";
+  version = "1.30.0";
```
Comment (lilyinstarlight) [inline pkgs/development/python-modules/meraki/default.nix:12]: I think we're starting to prefer adding `format` by default for all python packages, so it's probably best to make that explicit here:

```suggestion
  version = "1.30.0";
  format = "setuptools";
```

File: pkgs/development/python-modules/meraki/default.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    requests
+    setuptools
+  ] ++ typer.optional-dependencies.all;
+
+  pythonImportsCheck = [
+    "meraki"
+  ];
+
+  meta = with lib; {
+    description = "Provides all current Meraki dashboard API calls to interface with the Cisco Meraki cloud-managed platform.";
```
Comment (lilyinstarlight) [inline pkgs/development/python-modules/meraki/default.nix:30]:
```suggestion
    description = "Provides all current Meraki dashboard API calls to interface with the Cisco Meraki cloud-managed platform";
```

No period at the end of descriptions. See https://nixos.org/manual/nixpkgs/unstable/#var-meta-description

**Review by lilyinstarlight — CHANGES_REQUESTED** [review]
Thank you for your contribution! I've left a few comments to address to make it merge-ready

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (dylanmtaylor) [reply pkgs/top-level/all-packages.nix:9298]: There is a separate package that provides a CLI that I'm working on packaging. It needs this library to be packaged first though.

File: pkgs/development/python-modules/meraki/default.nix [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-s26xGwWSWB+qpOTUe8IYo53ywYOaaUWjDznFqpmRlak=";
+  };
+
+  propagatedBuildInputs = [
+    aiohttp
+    requests
+    setuptools
+  ] ++ typer.optional-dependencies.all;
```
Comment (lilyinstarlight) [inline pkgs/development/python-modules/meraki/default.nix:25]: The only dependencies I see actually referenced as required at runtime by `setup.py` are `aiohttp` and `requests`. Does this really need `setuptools` in `propagatedBuildInputs` and does it need the `typer` deps at all? (I don't see them referenced even in the development `requirements.txt` file in their repo)

**Review by lilyinstarlight — COMMENTED** [review]
Looks pretty good now, thank you!

I left one more small comment/question

File: pkgs/development/python-modules/meraki/default.nix (hunk 4 above)
Comment (dylanmtaylor) [reply pkgs/development/python-modules/meraki/default.nix:25]: I don't think so, it certainly just built without them. I think I copied this block from another package as a starter and just never removed those. I'll push a fix now.

File: pkgs/development/python-modules/meraki/default.nix [hunk 5]
```diff
@@ -0,0 +1,36 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, python3
+, aiohttp
+, requests
+, setuptools
+, typer
```
Comment (lilyinstarlight) [inline pkgs/development/python-modules/meraki/default.nix:8]:
```suggestion
, aiohttp
, requests
```

One last nit (not all bindings were used anymore) and then this PR looks good to me now!

Comment (dylanmtaylor) [reply pkgs/development/python-modules/meraki/default.nix]: Committed the suggestion then squashed to one commit. Thanks for the review!

**Review by lilyinstarlight — APPROVED** [review]
Looks awesome now! Once OfBorg finishes (just in case it catches something) I'll merge

---

Result of `nixpkgs-review pr 222268` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python310Packages.meraki</li>
    <li>python310Packages.meraki.dist</li>
    <li>python311Packages.meraki</li>
    <li>python311Packages.meraki.dist</li>
  </ul>
</details>

**Comment by dylanmtaylor** [comment]
> Looks awesome now! Once OfBorg finishes (just in case it catches something) I'll merge
> 
> Result of `nixpkgs-review pr 222268` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
> 
> 4 packages built:

Thank you so much. I'm guessing gnome-extensions-cli would be mergeable as well once that passes?

**Comment by lilyinstarlight** [comment]
> Thank you so much. I'm guessing gnome-extensions-cli would be mergeable as well once that passes?

I'll give it another review in a few minutes. It might still need the `typer` stuff removed and need unused args at the top removed too

**Comment by lilyinstarlight** [comment]
@ofborg build python310Packages.meraki python311Packages.meraki

Does ofborg not build python package set changes anymore or something 🤔? It didn't build any packages after eval

**Comment by dylanmtaylor** [comment]
> > Thank you so much. I'm guessing gnome-extensions-cli would be mergeable as well once that passes?
> 
> I'll give it another review in a few minutes. It might still need the `typer` stuff removed and need unused args at the top removed too

I went ahead and removed typer. The rest seem appear to be referenced/used in the file.

**Comment by lilyinstarlight** [comment]
The manually requested ofborg builds seem fine, so I'll go ahead and merge. Thanks again for your contribution!

**Pull request merged**